src/Stratis.Bitcoin/Base/ChainRepository.cs

```diff
@@ -18,17 +18,18 @@ public interface IChainRepository : IDisposable
         Task SaveAsync(ConcurrentChain chain);
     }
 
+    /// <summary>Provider of the last finalized block height.</summary>
+    /// <seealso cref="System.IDisposable" />
```

Comment:
Why would you need see also? This guy should not be disposable anyway, but even if it is, there should not be see also.

Also I would expect the concept of finalise the block to be explained somewhere. And by somewhere I probably mean here or on the implementation of the interface.

Reply:
Since we don't really have separated implementation of the interface, it should be here.

Reply:
Fixed